LocalyticsPlugin/Assets/Localytics/Localytics.cs

```diff
@@ -8,1073 +8,2163 @@
+			IntPtr set = AndroidJNI.CallObjectMethod(obj.GetRawObject(), keySetMethod, new jvalue[] { });
+			IntPtr toArrayMethod = AndroidJNI.GetMethodID(setClazz.GetRawClass(), "toArray", "()[Ljava/lang/Object;");
+			IntPtr array = AndroidJNI.CallObjectMethod(set, toArrayMethod, new jvalue[] { });
+
+			var dict = new Dictionary<string, string>();
+			var keys = AndroidJNIHelper.ConvertFromJNIArray<string[]>(array);
+			IntPtr getMethod = AndroidJNIHelper.GetMethodID(mapClazz.GetRawClass(), "get", "(Ljava/lang/Object;)Ljava/lang/Object;");
+			foreach (var k in keys)
+			{
+				string v = AndroidJNI.CallStringMethod(obj.GetRawObject(), getMethod, new jvalue[] { new jvalue() { l = AndroidJNI.NewStringUTF(k) } });
+				dict.Add(k, v);
+			}
+
+			return dict;
+		}
+
+		private static AndroidJavaObject ConvertCustomerInfo(CustomerInfo obj)
+		{
+			var builder = new AndroidJavaObject("com.localytics.android.Customer$Builder");
+			if (obj != null)
+			{
+				var args = new object[1];
+				if (obj.CustomerId != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setCustomerId", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.CustomerId;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.FirstName != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setFirstName", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.FirstName;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.LastName != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setLastName", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.LastName;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.FullName != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setFullName", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.FullName;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+
+				if (obj.EmailAddress != null)
+				{
+					IntPtr method = AndroidJNIHelper.GetMethodID(builder.GetRawClass(), "setEmailAddress", "(Ljava/lang/String;)Lcom.localytics.android.Customer$Builder;");
+					args[0] = obj.EmailAddress;
+					AndroidJNI.CallObjectMethod(builder.GetRawObject(), method, AndroidJNIHelper.CreateJNIArgArray(args));
+				}
+			}
+
+			return builder.Call<AndroidJavaObject> ("build", new object[]{});
+		}
+
+		private static List<CircularRegionInfo> ConvertGeofencesList(AndroidJavaObject listObject)
+		{
+			int size = listObject.Call<int>("size");
```

Comment:
Yes, we own the APIs that will result in this conversion. It only gets called from:
```
public static List<CircularRegion> getGeofencesToMonitor(double latitude, double longitude);
void localyticsDidTriggerRegions(@NonNull List<Region> regions, @NonNull Region.Event event);
void localyticsDidUpdateMonitoredGeofences(@NonNull List<CircularRegion> added, @NonNull List<CircularRegion> removed);
```